app/views/support/WeeklyPromotion.scala

```diff
@@ -41,38 +46,66 @@ object WeeklyPromotion {
         discountedPlans = plansForPromotion(promotion, promoCode, ZONEC - requestCountry, catalog.weeklyZoneC.toList)
       )
     }
-    val UKandUS = Seq(DiscountedRegion(
-      title = "United Kingdom",
-      description = "Includes Isle of Man and Channel Islands",
-      countries = UK,
-      discountedPlans = plansForPromotion(promotion, promoCode, UK, catalog.weeklyZoneA.toList)
-    ), DiscountedRegion(
+    val UKregion: Set[DiscountedRegion] = {
+      val all = DiscountedRegion(
+        title = "United Kingdom",
+        description = "Includes Isle of Man and Channel Islands",
+        countries = UK,
+        discountedPlans = plansForPromotion(promotion, promoCode, UK, catalog.weeklyZoneA.toList)
+      )
+      val domestic = DiscountedRegion(
+        title = "United Kingdom",
+        description = "Includes mainland UK only.",
+        countries = UKdomestic,
+        discountedPlans = plansForPromotion(promotion, promoCode, UK, catalog.weeklyZoneA.toList)
+      )
+      val overseas = DiscountedRegion(
+        title = "Isle of Man and Channel Islands",
+        description = "",
+        countries = UKoverseas,
+        discountedPlans = plansForPromotion(promotion, promoCode, UK, catalog.weeklyZoneA.toList)
+      )
+      val countries = promotionCountries intersect UK
```

Comment:
do we need to intersect with uk first if we're then going to intersect with domestic uk and non domestic uk?